BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -1195,6 +1238,67 @@ private bool GunFiringSpeedCheck()
                 return relVelSqrMag < firingSpeed * firingSpeed;
         }
 
+        private Vector3 GunFiringSolution(ModuleWeapon weapon)
+        {
+            // For fixed weapons, returns attitude that puts fixed weapon on target, even if not aligned with vesselTransform.up
+            // For turreted weapons, returns attitude toward target or broadside attitude depending on BroadsideAttack setting
+
+            Vector3 firingSolution;
+            if (weapon != null && !weapon.turret)
+            {
+                Vector3 leadOffset = weapon.GetLeadOffset();
+                Vector3 target = targetVessel.CoM;
+                target -= leadOffset;  // Lead offset from aiming assuming the gun is forward aligned and centred.
+                                       // Note: depending on the airframe, there is an island of stability around -2°—30° in pitch and ±10° in yaw where the vessel can stably aim with offset weapons.
+                Vector3 weaponPosition = weapon.fireTransforms[0].position;
```

Comment:
If we're going to start using this in multiple places, then it might be worth moving the calculation of `weaponPosition` and `weaponDirection` relative to `vessel.ReferenceTransform.position` to a common function in `ModuleWeapon` and have it only update on weapon selection so that it's not being recalculated each frame.